pkg/storage/secret/metadata/keeper_store.go

```diff
@@ -230,21 +327,47 @@ func (s *keeperMetadataStorage) List(ctx context.Context, namespace xkube.Namesp
 }
 
 // validateSecureValueReferences checks that all secure values referenced by the keeper exist and are not referenced by other third-party keepers.
-func (s *keeperMetadataStorage) validateSecureValueReferences(sess *sqlstore.DBSession, keeper *secretv0alpha1.Keeper) error {
+func (s *keeperMetadataStorage) validateSecureValueReferences(ctx context.Context, sess *session.SessionTx, keeper *secretv0alpha1.Keeper) error {
 	usedSecureValues := extractSecureValues(keeper)
 
 	// No secure values are referenced, return early.
 	if len(usedSecureValues) == 0 {
 		return nil
 	}
 
-	secureValueCond := &secureValueDB{Namespace: keeper.Namespace}
-	secureValueRows := make([]*secureValueDB, 0)
+	reqSecValue := listByNameSecureValue{
+		SQLTemplate:      sqltemplate.New(s.dialect),
+		Namespace:        keeper.Namespace,
+		UsedSecureValues: usedSecureValues,
+	}
 
-	// SELECT * FROM secret_secure_value WHERE name IN (...) AND namespace = ? FOR UPDATE;
-	err := sess.Table(secureValueCond.TableName()).ForUpdate().In("name", usedSecureValues).Find(&secureValueRows, secureValueCond)
+	qSecValue, err := sqltemplate.Execute(sqlSecureValueListByName, reqSecValue)
 	if err != nil {
-		return fmt.Errorf("check securevalues existence: %w", err)
+		return fmt.Errorf("list template %q: %w", qSecValue, err)
+	}
+
+	rows, err := sess.Query(ctx, qSecValue, reqSecValue.GetArgs()...)
+	if err != nil {
+		return fmt.Errorf("list template %q: %w", qSecValue, err)
```

Comment:
Something like this, I don't think we should log the whole sql query 🤔 
```suggestion
		return fmt.Errorf("executing query: %w", err)
```